noir_stdlib/src/field.nr

```diff
@@ -15,6 +15,14 @@ impl Field {
     #[builtin(to_be_bits)]
     fn __to_be_bits(_self: Self, _bit_size: u32) -> [u1] {}
 
+    #[builtin(apply_range_constraint)]
+    fn __assert_maximum_bit_size(_self: Self, _bit_size: u32) {}
+
+    pub fn assert_maximum_bit_size(self: Self, bit_size: u32) {
+        crate::assert_constant(bit_size);
+        self.__assert_maximum_bit_size(bit_size);
```

Comment:
```suggestion
        self.__assert_maximum_bit_size(bit_size);
        assert(bit_size < modulus_num_bits())
```

Let's add an assertion here to check that the range check is meaningful. This also allows us to safely unwrap this into a `u32` without having to potentially return an error.